src/main/java/beam/analysis/plots/RealizedModeAnalysis.java

```diff
@@ -390,6 +417,35 @@ public void writeToRootCSV() {
     }
 
 
+    private void writeToReplanningCSV(IterationEndsEvent event){
+        String fileName = GraphsStatsAgentSimEventsListener.CONTROLLER_IO.getIterationFilename(event.getIteration(), "replanningCountModeChoice.csv");
+        try{
```

Comment:
You didn't close. How about using try disposable?

    try(BufferedWriter out = new BufferedWriter(new FileWriter(new File(fileName)))) {

Then you also don't have to worry about flushing either.

Reply:
yes